seqio/docs/source/overview.md

```diff
@@ -0,0 +1,962 @@
+  * `seed`: An optional seed to use for deterministic shuffling and (stateless) stochastic ops. These operations will still be pseudorandom but will be reproducible with the same seed. Set to `None` if determinism is not desired.
+
+### (Optional) Offline Caching
+
+For improved performance at load time and avoid redundant computations for commonly used tasks, you can pre-cache your `Task` with all or part of the preprocessing done in advance of training.
+
+The first step to doing so is to add a `seqio.CacheDatasetPlaceholder(required=False)` as one of the steps in your preprocessing pipeline. All steps before the placeholder will be cached offline and all steps after will be executed on the fly at load time. You may set `required=True` if you want `get_dataset` to fail unless `use_cached=True`.
+
+Caveats:
+
+* Any stochastic operations that you wish to be re-run when `num_epochs > 1` or with a different `seed` *should* go after the placeholder since only a single sample will be cached.
+* Any preprocessing steps that use the `sequence_length` argument *must* come after the `seqio.CacheDatasetPlaceholder` preproessor since this is only known at runtime, or an exception will be raised. If you wish to cache for a specific sequence length, you can use [`seqio.experimental.add_fully_cached_task`](http://google3/third_party/py/seqio/experimental.py).
+
+Once your `Task` is registered, you can run [`cache_tasks_main`](scripts/cache_tasks_main.py) to execute the offline preprocessing, providing it with the module containing your task definitions via the `--module_import` flag. For very large datasets, it's recommended you run this [Apache Beam](https://beam.apache.org/) script on a distributed framework like [Google Cloud DataFlow](https://beam.apache.org/documentation/runners/dataflow/).
+
+Finally, you are ready to load the cached version of your `Task` (or `Mixture`) containing it. You will need to add the path to the directory you passed to `--output_cache_dir` via `seqio.add_global_cache_dirs(["/my/cache/dir"])`. Now when you call `task_or_mixture.get_dataset(..., use_cached=True)`, the data will be loaded from the cache directory instead of the raw data source.
+
+### Feature Converters
+
+The role of `Task` is to provide the dataset object with as little
+model-specific features (e.g., generic "inputs" and "targets") while the Feature
+Converters transform the model-agnostic features to model-specific features
+(e.g., "encoder_input_tokens"). We refer to the former as "task features" and
+the latter as "model features".
+
+
+Let's use machine translation (English to German) as a running example.
+
+The raw data consists of sentence pairs such as
+
+```
+"That is good\tDas ist gut."
+```
+
+A task registered to `Task` (e.g.,
+[wmt_t2t_ende_v003](http://google3/third_party/py/t5/data/tasks.py?l=156&rcl=337594707))
```

Comment:
change to https://github.com/google-research/text-to-text-transfer-transformer/blob/main/t5/data/tasks.py#L212